### DISCUSSION TITLE
feat: add search panel for project wide search

### DISCUSSION THREAD

**Discussion by CarreraPHP** [zed-industries/zed#7206] [Feature Requests]
Please add Search Panel for searching contents of files in the entire project similar to how VSCode does.

I mostly work in large migration projects where we need to rewrite code of entire repo from one technology to another while the Business logic are untouched. 

To exemplify, i am listing some of the migrations done in the past
- Redux + Saga to Apollo Graphql
- Jest to Vitest
- Webpack Module Fedaration with Monorepo
- Webpack to Vite

Migrations of above nature, would require us rewrite code in alot of files. for example the vitest mocking implementation is async compared to jest. which means, you have find all the files in which the development teams have written their mocks and make the modifications. The big challenge in migrations is TIME. Management would expect things should get completed in shortest time possible.

VSCode's Search & Replace feature with Regex support was immensely helpful in rewriting the code in shortest time possible. Maybe, this feature can be worked out from ground-up using the AI and awareness of the AST/CST of the files.

To start with a minimum search using keyword across project would be of great help.

**Comment by SomeoneToIgnore** [comment]
Would `workspace: new search` (cmd-shift-f by default) work for you as the tool needed?
![image](https://github.com/zed-industries/zed/assets/2690773/64406354-56b6-48b9-a484-007d78632a5f)

Both search, regex and replace functions are supported in general, but may have bugs.

**Comment by CarreraPHP** [comment]
@SomeoneToIgnore , Thanks for sharing. wouldn't be great if there is a search icon next to the project panel icon that can bring the search tab.

**Comment by Vanekkk1** [comment]
I need an icon for search, only buttons are not convenient, like a loop icon in vs code
